src/codegen/codegen_compatibility_visitor.cpp

```diff
@@ -0,0 +1,142 @@
+/*************************************************************************
+ * Copyright (C) 2018-2019 Blue Brain Project
+ *
+ * This file is part of NMODL distributed under the terms of the GNU
+ * Lesser General Public License. See top-level LICENSE file for details.
+ *************************************************************************/
+
+#include <algorithm>
+#include <iostream>
+#include <memory>
+#include <string>
+
+#include "codegen/codegen_compatibility_visitor.hpp"
+#include "utils/logger.hpp"
+#include "visitors/lookup_visitor.hpp"
+
+namespace nmodl {
+namespace codegen {
+
+using visitor::AstLookupVisitor;
+
+/**
+ * \details Checks all the ast::AstNodeTypes that are not compatible with NMODL code
+ * generation and prints related messages. If there is some kind of incompatibility
+ * stop NMODL code generation.
+ */
+bool CodegenCompatibilityVisitor::find_incompatible_ast_nodes(Ast* node) {
+    std::vector<AstNodeType> incompatible_ast_types = {AstNodeType::SOLVE_BLOCK,
```

Comment:
Good point to move it to the header file, however it needs to be a vector since it is used by `AstLookupVisitor::lookup(Ast* node, std::vector<AstNodeType>& _types)` that takes as second argument a `vector` of `AstNodeType`s

Reply:
Yes, sorry! my bad. I realized later but forgot to update my comment.